sbt-dump-event-schema/src/main/scala/org/virtuslab/ash/DumpEventSchema.scala

```diff
@@ -1,37 +1,40 @@
 package org.virtuslab.ash
 
 import better.files._
+import net.jcazevedo.moultingyaml.PimpedAny
+import net.jcazevedo.moultingyaml.YamlArray
+import net.jcazevedo.moultingyaml.YamlObject
 import sbt.Keys._
 import sbt.{File => _, _}
 import spray.json._
 
-object DumpEventSchema extends DefaultJsonProtocol {
+import org.virtuslab.ash.model._
+
+object DumpEventSchema {
+  import DumpEventSchemaJsonProtocol.typeDefinitionFormatJson
+  import DumpEventSchemaYamlProtocol._
   import DumpEventSchemaPlugin.autoImport._
 
   def apply(outputFile: File, inputDirectory: File): Unit = {
-    val jsons = for {
-      file <- inputDirectory.list.toSeq.sorted(File.Order.byName)
-    } yield new String(file.loadBytes).parseJson.asJsObject
+    val typeDefinitions = for {
+      file <- inputDirectory.list.filterNot(_.isDirectory)
+    } yield new String(file.loadBytes).parseJson.convertTo[TypeDefinition]
```

Comment:
Hmmm would it be possible/reasonable to use YAML everywhere, also for the intermediate files?

Reply:
Not with current yaml libraries. `circle-yaml` depends on `circe` which in turn depends on `cats`. Compiler plugin is an uber jar, meaning we would add `cats` to every compilation run that uses our plugin.